autogen/generator.py

```diff
@@ -320,11 +320,16 @@ def __call__(self):
         D = sorted(D.values(), key=lambda d: d['function'])
         sys.stdout.write("Generating PARI functions:")
 
-        self.gen_file = open(self.gen_filename + '.tmp', 'w')
+        # Stupid Python 3 forces us to specify an encoding
+        if "encoding" in open.__doc__:
+            kwds = dict(mode="wt", encoding="utf-8")
+        else:
+            kwds = dict(mode="wt")
```

Comment:
The "cross-python" compatible way to write this:

```
import io
f = io.open(filename, 'w', encoding='utf-8')
```

if you want to specify UTF-8.  On Python 3, `builtins.open` *is* `io.open`.  On Python 2 it is not (`open` is the legacy built-in from before the I/O refactoring, which in Python 2 is fully encapsulated in the `io` module).

For Python2/3 compatibility w.r.t. I/O it's almost always best to use the `io` module explicitly.